lib/drowsy/scoping.rb

```diff
@@ -6,14 +6,20 @@ module Drowsy::Scoping
   class_methods do
     delegate(
       :find, :find_by, :find_by!,
+      :all,
       :where,
       :build,
       :create, :create!,
-      to: :all
+      to: :blank_relation
     )
 
-    def all
-      Drowsy::Relation.new(self)
+    def scope(name, behavior)
+      define_singleton_method name, behavior
+      blank_relation.define_singleton_method(name, behavior)
+    end
+
+    def blank_relation
+      @blank_relation ||= Drowsy::Relation.new(self)
     end
```

Comment:
@smudge so, i learned that this strategy of defining a singleton method on a blank relation instance only works because this instance is memoized AND the chaining in Relation uses `clone` to produce new instances. `clone` maintains the singleton class.

Reply:
Yeah -- `clone` is key.